js/tweek-local-cache/src/tweek-repository.ts

```diff
@@ -74,7 +75,7 @@ export default class TweekRepository {
       keys = keys || {};
       Object.entries(keys).forEach(([key, value]) => {
         if (value.expiration) {
```

Comment:
don't we also need to check if a key state is `requested` and run `this._dirtyRefresh()`?

Reply:
can you elaborate?  I did this under the assumption that `_isDirty` is the only flag that should initiate a refresh (since, if it's false, the promise is just immediately resolves and no refresh occurs).

Reply:
in the `_refreshKeys` function we get all the keys that are either requested or expired https://github.com/Soluto/tweek-clients/blob/2f65f9511b8e12948d26dc7649e69e2f2e760f25/js/tweek-local-cache/src/tweek-repository.ts#L240
because it used to run all the time, it should have fetched the requested keys as well.
if it didn't then we had a bug :)

Reply:
right, but we don't get to that line if `_isDirty` is false, so I don't think I've changed the existing functionality that was in place.  Perhaps I am missing something though?

Reply:
you are right, i think we had a bug that i noticed just now.
maybe add a test case for it and see if it's a real bug.
when doing `useStore` with keys that have `requested` state but don't have `expired` set and see if they get refreshed

Reply:
i just checked and we only do store.save after _updateTrieKeys which changes the cache's state to something that will never be 'requested', so I think we're good not needing to check it in useStore